src/Bicep.Core/Configuration/RootConfiguration.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Bicep.Core.Configuration
+{
+    public class RootConfiguration
+    {
+        private RootConfiguration(CloudConfiguration cloud, ModuleAliasesConfiguration moduleAliases, AnalyzersConfiguration analyzers, string resourceName)
+        {
+            this.Cloud = cloud;
+            this.ModuleAliases = moduleAliases;
+            this.Analyzers = analyzers;
+            this.ResourceName = resourceName;
+        }
```

Comment:
`RootConfiguration` is immutable, right? If the underlying `IConfiguration` changes, does that affect the bound values?

Reply:
It will, that's why `IConfiguration` is wrapped by `AnalyzerConfiguration` which only exposed a `GetValue` method.